File: scripts/build-thumbs.py
#!/usr/bin/env python3
"""Regenerate portal card thumbnails from the meta-image set.

The meta-images have each template's TITLE + preview + features baked in, so the
mosaic wall reads without hover. Local dev tool; the resulting .webp are COMMITTED
to public/thumbs/ so the standalone Vercel build needs no sibling assets.

Run:  python3 scripts/build-thumbs.py   (or: pnpm thumbs)
"""
from pathlib import Path
from PIL import Image

PORTAL = Path(__file__).resolve().parents[1]
ROOT = PORTAL.parent  # _templates root holding meta-image/
META = ROOT / "meta-image"
OUT = PORTAL / "public" / "thumbs"
OUT.mkdir(parents=True, exist_ok=True)

# slug -> meta-image file (titles baked in; mapped by visual inspection 2026-06-21)
MAP = {
    "personal-brand-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (1).png",
    "agency-studio-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (2).png",
    "konsultan-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (3).png",
    "kreator-studio-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (4).png",
    "riset-kit": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (5).png",
    "saas-marketing-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (6).png",
    "wirausaha-os": "ChatGPT Image Jun 4, 2026, 04_30_15 PM (7).png",
}
W = 1600
missing = []
for slug, fn in MAP.items():
    src = META / fn
    if not src.exists():
        missing.append(fn)
        continue
    im = Image.open(src).convert("RGB")
    h = round(im.height * W / im.width)
    im.resize((W, h), Image.LANCZOS).save(OUT / f"{slug}.webp", "WEBP", quality=82, method=6)
    print(f"thumb  {slug}.webp  {W}x{h}")
print(f"\n{len(MAP) - len(missing)}/{len(MAP)} thumbs built" + (f" — MISSING: {missing}" if missing else " ✓"))

File: scripts/build-thumbs.mjs
// Regenerate portal card thumbnails from each template's public/hero.webp.
//
// Local dev tool only: the resulting thumbs are COMMITTED to public/thumbs/, so
// the standalone Vercel build of this repo needs no sibling template repos. Run
// after a template's hero changes:  node scripts/build-thumbs.mjs  (or: pnpm thumbs)
//
// Slugs are read from the SSOT registry (data/templates.ts) so this stays in
// sync automatically. Source = ../<slug>/public/hero.webp in the _templates
// monorepo checkout.
import { readFileSync, copyFileSync, mkdirSync, existsSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

const here = dirname(fileURLToPath(import.meta.url));
const portal = join(here, "..");
const reposRoot = join(portal, ".."); // _templates root holding the sibling templates
const thumbs = join(portal, "public", "thumbs");

const registry = readFileSync(join(portal, "data", "templates.ts"), "utf8");
const slugs = [...registry.matchAll(/slug:\s*"([^"]+)"/g)].map((m) => m[1]);

mkdirSync(thumbs, { recursive: true });
let ok = 0;
const missing = [];
for (const slug of slugs) {
  const src = join(reposRoot, slug, "public", "hero.webp");
  if (!existsSync(src)) {
    missing.push(slug);
    continue;
  }
  copyFileSync(src, join(thumbs, `${slug}.webp`));
  ok++;
  console.log(`thumb  ${slug}.webp`);
}
console.log(
  `\n${ok}/${slugs.length} thumbs built` +
    (missing.length ? ` — missing hero.webp for: ${missing.join(", ")}` : " ✓"),
);
if (missing.length) process.exitCode = 1;
